oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "../sandboxed.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/sandbox2/util/path.h"
+#include "tiffio.h"
+
+/* sapi functions:
+ *    TIFFTileSize
+ *    TIFFOpen
+ *    TIFFReadEncodedTile
+ *    TIFFSetField
+ *    TIFFClose
+ *    TIFFReadRGBATile
+ *    TIFFGetField
+ */
+
+static const std::vector<unsigned char> cluster_0 = {0, 0, 2, 0, 138, 139};
+static const std::vector<unsigned char> cluster_64 = {0, 0, 9, 6, 134, 119};
+static const std::vector<unsigned char> cluster_128 = {44, 40, 63, 59, 230, 95};
+
+static int CheckCluster(int cluster,
+                        const sapi::v::Array<unsigned char> &buffer,
+                        const std::vector<unsigned char> &expected_cluster) {
+  unsigned char* target = buffer.GetData() + cluster * 6;
+
+  if (!std::memcmp(target, expected_cluster.data(), 6)) {
+    return 0;
+  }
+
+  std::cerr << "Cluster " << cluster << " did not match expected results.\n"
+            << "Expect: " << expected_cluster[0] << "\t" << expected_cluster[1]
+            << "\t" << expected_cluster[4] << "\t" << expected_cluster[5]
+            << "\t" << expected_cluster[2] << "\t" << expected_cluster[3]
+            << "\n"
+            << "Got: " << target[0] << "\t" << target[1] << "\t" << target[4]
+            << "\t" << target[5] << "\t" << target[2] << "\t" << target[3]
+            << '\n';
+
+  return 1;
+}
```

Comment:
This function should return an `absl::Status` instead and include the error message there. Same for the other functions that can return a failure with a reason.

Use `absl::UnknownError()` or `absl::InternalError()` to construct these.